File: Data Structures & Algorithms/eating-bananas/submission-0.cpp
class Solution {
public:
    int minEatingSpeed(vector<int>& piles, int h) {
        int l = 1;  // eating speed must be >= 1
        int r = *max_element(piles.begin(), piles.end());
        int res = r;

        while (l <= r) {
            int mid = l + (r - l) / 2;
            long long hours = 0;

            for (int p : piles) {
                hours += (p + mid - 1) / mid; // integer ceil
            }

            if (hours <= h) {   // ✅ check against h
                res = mid;
                r = mid - 1;    // try smaller
            } else {
                l = mid + 1;    // need bigger
            }
        }
        return res;
    }
};

File: Data Structures & Algorithms/subsets/submission-0.java
class Solution {
    public List<List<Integer>> subsets(int[] nums) {
        List<List<Integer>>res=new ArrayList<>();
        
        res.add(new ArrayList<>());
        int l=0;
        int r=nums.length;
        for(int i =0;i<nums.length;i++){
            for(int j=0;j<nums.length;j++){
                List<Integer>temp=new ArrayList<>(res.get(i));
                temp.add(nums[i]);
                res.add(temp);
            }
            
        }
        return res;
        
        
        
    }
}
